codec-ohttp-hpke-bouncycastle/src/main/java/io/netty/incubator/codec/hpke/bouncycastle/BouncyCastleOHttpCryptoProvider.java

```diff
@@ -184,6 +185,26 @@ private static ECDomainParameters ecDomainParameters(KEM kem) {
         }
     }
 
+    @Override
+    public AsymmetricCipherKeyPair newRandomPrivateKey(KEM kem) {
+        return new BouncyCastleAsymmetricCipherKeyPair(newRandomPair(kem));
+    }
+
+    private static org.bouncycastle.crypto.AsymmetricCipherKeyPair newRandomPair(KEM kem) {
+        switch (kem) {
+            case X25519_SHA256:
+                X25519PrivateKeyParameters x25519PrivateKey = new X25519PrivateKeyParameters(new SecureRandom());
+                return new org.bouncycastle.crypto.AsymmetricCipherKeyPair(
+                        x25519PrivateKey.generatePublicKey(), x25519PrivateKey);
+            case X448_SHA512:
+                X448PrivateKeyParameters x448PrivateKey = new X448PrivateKeyParameters(new SecureRandom());
```

Comment:
@thomdev I did decide to generate a new SecureRandom key every time. This is expensive so I wonder if we should just use the same instance all the time ?